source/core/src/main/com/csse3200/game/screens/MainGameScreen.java

```diff
@@ -126,30 +66,40 @@ public MainGameScreen(GdxGame game) {
         ServiceLocator.registerAlertBoxService(new AlertBoxService(stage, skin));
 
         // todo load based on what the user chose
-        boolean shouldLoad = gameOptions.shouldLoad;
         logger.info("Should start from save file: {}", shouldLoad);
 
         /*
          * based on the characters selected, changed the link
          * If Player choose Load, then create
          */
         this.playerFactory = new PlayerFactory(Arrays.stream(PLAYERS).toList());
-        Entity player = loadPlayer(shouldLoad, gameOptions, chosenPlayer);
+        Entity player = loadPlayer(shouldLoad, gameOptions, chosenPlayer, gameOptions.difficulty);
 
         player.getEvents().addListener("player_finished_dying", this::loseGame);
 
         logger.debug("Initialising main game screen entities");
-
-        LevelFactory levelFactory = new MainGameLevelFactory(shouldLoad);
-
+        MapLoadConfig mapConfig= new MapLoadConfig(); 
+        mapConfig.currentLevel = "0";
+        LevelFactory levelFactory = new MainGameLevelFactory(shouldLoad, mapConfig);
         if (gameOptions.difficulty == TEST) {
             new TestGameArea(levelFactory, player);
         } else {
-            new MainGameArea(levelFactory, player, shouldLoad);
+            new MainGameArea(levelFactory, player, shouldLoad, mapConfig);
         }
     }
 
-    private Entity loadPlayer(boolean shouldLoad, GameOptions gameOptions, String chosenPlayer) {
+    /**
+     * Loads map data from the save file and extracts the completed rooms into a list which
+     * is later set to completed in the create method above
+     * @param filePath: Path for the save file
+     */
+    public MapLoadConfig loadFromJson(String filePath) {
+        MapLoadConfig mapLoadConfig = new MapLoadConfig();
+        mapLoadConfig  = FileLoader.readClass(MapLoadConfig.class, filePath, FileLoader.Location.EXTERNAL);
+        return mapLoadConfig;
+    }
```

Comment:
this is unused?

Reply:
deleted it.. we had it just in case if we needed to refer back to it but we changed the implementation now.